src/test/java/org/apache/commons/crypto/stream/CtrCryptoStreamTest.java

```diff
@@ -176,14 +162,8 @@ protected void doDecryptTest(final String cipherClass, final boolean withChannel
         final byte[] expectedData = new byte[dataLen];
         buf.get(readData);
         System.arraycopy(data, 0, expectedData, 0, dataLen);
-        Assert.assertArrayEquals(readData, expectedData);
-
-        try {
-            in.decryptBuffer(buf);
-            Assert.fail("Expected IOException.");
-        } catch (final IOException ex) {
-            Assert.assertEquals(ex.getCause().getClass(), ShortBufferException.class);
-        }
+        assertArrayEquals(readData, expectedData);
+        assertThrows(IOException.class, () -> in.decryptBuffer(buf));
```

Comment:
Again, you've lost some of the testing. In this case, the test checks for ShortBufferException.

I'm going to pause my review of this PR until you revisit all changes to make sure you keep/restore ALL the test assertions.